subsys/bluetooth/nrf_bt_adv/providers/tx_power.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/sys/byteorder.h>
+
+#include <zephyr/bluetooth/hci.h>
+#include <zephyr/bluetooth/hci_vs.h>
+
+#include <bluetooth/nrf_bt_adv.h>
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(nrf_bt_adv, CONFIG_NRF_BT_ADV_LOG_LEVEL);
+
+
+static int get_data(struct bt_data *ad, const struct nrf_bt_adv_state *state,
+		    struct nrf_bt_adv_feedback *fb)
+{
+	ARG_UNUSED(state);
+	ARG_UNUSED(fb);
+
+	static uint8_t tx_power;
+
+	struct bt_hci_cp_vs_read_tx_power_level *cp;
+	struct bt_hci_rp_vs_read_tx_power_level *rp;
+	struct net_buf *rsp = NULL;
+	struct net_buf *cmd_buf = bt_hci_cmd_create(BT_HCI_OP_VS_READ_TX_POWER_LEVEL, sizeof(*cp));
+
+	if (!cmd_buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(cmd_buf, sizeof(*cp));
+	cp->handle_type = BT_HCI_VS_LL_HANDLE_TYPE_ADV;
+	cp->handle = sys_cpu_to_le16(0);
+
+	int err = bt_hci_cmd_send_sync(BT_HCI_OP_VS_READ_TX_POWER_LEVEL, cmd_buf, &rsp);
+
+	if (err) {
+		LOG_ERR("Read Tx power err: %d", err);
+	} else {
+		rp = (struct bt_hci_rp_vs_read_tx_power_level *)rsp->data;
+		tx_power = (uint8_t)rp->tx_power_level;
```

Comment:
You can consider checking rp->status before setting tx_power. I assume that `bt_hci_cmd_send_sync` can return with success but response status can be set to error.

Reply:
Looking at the implementation non-zero status code in the response always leads to returning an error:
https://github.com/nrfconnect/sdk-zephyr/blob/main/subsys/bluetooth/host/hci_core.c#L305

The API doc (https://docs.zephyrproject.org/apidoc/latest/hci_8h.html#a8d306b5b56afdece1cadbbf0eecfe5a7) also states that:
`The function will block until a Command Status or a Command Complete event is returned. If either of these have a non-zero status the function will return a negative error code and the response reference will not be set.`

So that I think checking the status is unnecessary if err == 0